timemachine/md/minimizer.py

```diff
@@ -482,7 +482,7 @@ def val_and_grad_fn_local(x_local):
         u_full, grad_full = val_and_grad_fn(x_prime)
         # avoid being trapped when overflows spuriously appear as large negative numbers
         # remove after resolution of https://github.com/proteneer/timemachine/issues/481
-        if u_0 - u_full > guard_threshold:
+        if not np.isfinite(u_full) or u_0 - u_full > guard_threshold:
```

Comment:
Is `guard_threshold` still relevant here, with the added overflow handling?

Reply:
Yes, you can still add N energies of `MAX_LONG_LONG - 1` that will overflow in a way that is not account for. There will be a follow-up PR that handles that case. There is a failing test of this latter case: https://github.com/proteneer/timemachine/blob/feature/detect-nb-clashes/tests/test_energy_overflows.py#L249-L282